### PR TITLE
[Feature]: Support tensor2grayimgs

### PR DESCRIPTION
**Pull request by cuhk-hbsun**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support tensor2imgs while tensor with 1 channel.

## Modification

Update the `assert` check in function `tensor2imgs` under `mmcv/image/misc.py`.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by zhouzaida** [comment]
Please merge the upstream master to fix the error of building docs.

**Comment by zhouzaida** [comment]
Please @gaotongxiao have a look.

File: tests/test_image/test_image_misc.py [hunk 1]
```diff
@@ -24,15 +24,29 @@ def test_tensor2imgs():
...
+    with pytest.raises(AssertionError):
+        tensor = torch.randn(2, 4, 3, 3)
+        mmcv.tensor2imgs(tensor)
+
     # test mean length
     with pytest.raises(AssertionError):
         tensor = torch.randn(2, 3, 5, 5)
         mmcv.tensor2imgs(tensor, mean=(1, ))
+        tensor = torch.randn(2, 1, 5, 5)
+        mmcv.tensor2imgs(tensor)
```
Comment (gaotongxiao) [inline tests/test_image/test_image_misc.py:37]: Would it be better if the dimensionality of the default mean and std automatically matches the input channels?

File: mmcv/image/misc.py [hunk 2]
```diff
@@ -9,15 +9,18 @@
...
     """Convert tensor to 3-channel images or 1-channel gray images.
 
     Args:
         tensor (torch.Tensor): Tensor that contains multiple images, shape (
             N, C, H, W). :math:`C` can be either 3 or 1.
-        mean (tuple[float], optional): Mean of images. Defaults to (0, 0, 0).
-        std (tuple[float], optional): Standard deviation of images.
-            Defaults to (1, 1, 1).
+        mean (tuple[float], optional): Mean of images. If None, default one
+            will be used according to the channel dim of tensor.
```
Comment (gaotongxiao) [inline mmcv/image/misc.py:19]: "default one" is ambiguous. Suggested change: "`(0, )` or `(0, 0, 0)` will be used..."

File: mmcv/image/misc.py [hunk 3]
```diff
@@ -9,15 +9,18 @@
...
+            Defaults to None.
+        std (tuple[float], optional): Standard deviation of images. If None,
+            default one will be used according to the channel dim of tensor.
```
Comment (gaotongxiao) [inline mmcv/image/misc.py:22]: same

**Review by teamwong111 — APPROVED** [review]
LGTM

**Review by gaotongxiao — APPROVED** [review]

File: mmcv/image/misc.py [hunk 4]
```diff
@@ -29,8 +32,13 @@ def tensor2imgs(tensor, mean=(0, 0, 0), std=(1, 1, 1), to_rgb=True):
     if torch is None:
         raise RuntimeError('pytorch is not installed')
     assert torch.is_tensor(tensor) and tensor.ndim == 4
-    assert len(mean) == 3
-    assert len(std) == 3
+    assert tensor.size(1) in [1, 3]
+    if mean is None:
+        mean = (0, ) if tensor.size(1) == 1 else (0, 0, 0)
```
Comment (zhouzaida) [inline mmcv/image/misc.py:37]:
```suggestion
        mean = (0, ) * tensor.size(1)
```

File: mmcv/image/misc.py [hunk 5]
```diff
@@ -29,8 +32,13 @@ def tensor2imgs(tensor, mean=(0, 0, 0), std=(1, 1, 1), to_rgb=True):
...
+    if std is None:
+        std = (1, ) if tensor.size(1) == 1 else (1, 1, 1)
```
Comment (zhouzaida) [inline mmcv/image/misc.py:39]:
```suggestion
        std = (1, ) * tensor.size(1)
```

File: mmcv/image/misc.py [hunk 6]
```diff
@@ -29,8 +32,13 @@ def tensor2imgs(tensor, mean=(0, 0, 0), std=(1, 1, 1), to_rgb=True):
...
+    assert (tensor.size(1) == len(mean) == len(std) == 3) or \
+        (tensor.size(1) == len(mean) == len(std) == 1 and not to_rgb)
```
Comment (zhouzaida) [inline mmcv/image/misc.py:41]:
```suggestion
    channels = tensor.size(1)
    assert channels in [1, 3]
    if mean is None:
        mean = (0, ) * channels
    if std is None:
        std = (1, ) * channels
    assert (channels == len(mean) == len(std)) or \
        (channels == len(mean) == len(std) and not to_rgb)
```

**Review by zhouzaida — APPROVED** [review]

File: mmcv/image/misc.py [hunk 7]
```diff
@@ -29,8 +32,14 @@ def tensor2imgs(tensor, mean=(0, 0, 0), std=(1, 1, 1), to_rgb=True):
     if torch is None:
         raise RuntimeError('pytorch is not installed')
     assert torch.is_tensor(tensor) and tensor.ndim == 4
-    assert len(mean) == 3
-    assert len(std) == 3
+    channels = torch.size(1)
```
Comment (zhouzaida) [inline mmcv/image/misc.py:35]:
```suggestion
    channels = tensor.size(1)
```

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**